manifest/manifest.go

```diff
@@ -0,0 +1,150 @@
+package manifest
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/ipfs/go-datastore"
+	"golang.org/x/xerrors"
+)
+
+type OnManifestChange func(ctx context.Context, initialInstance uint64, rebootstrap bool, errCh chan error)
+
+type ManifestProvider interface {
+	// Run starts any background tasks required for the operation
+	// of the manifest provider.
+	Run(context.Context, chan error)
+	// Returns the list of gpbft options that should be used
+	GpbftOptions() []gpbft.Option
+	// suscribe to manifest updates
+	Subscribe() <-chan struct{}
+	// Set callback for manifest changes
+	SetManifestChangeCb(OnManifestChange)
+
+	// Manifest accessors
+	MsgPubSubTopic() string
+	NetworkName() gpbft.NetworkName
+	BootstrapEpoch() int64
+	DatastorePrefix() datastore.Key
+	InitialPowerTable() []gpbft.PowerEntry
+	EcConfig() *EcConfig
+}
+
+type Version string
+
+type GpbftConfig struct {
+	Delta                time.Duration
+	DeltaBackOffExponent float64
+	MaxLookaheadRounds   uint64
+}
+
+type EcConfig struct {
+	ECFinality       int64
+	ECDelay          time.Duration
+	CommiteeLookback uint64
+
+	//Temporary
+	ECPeriod            time.Duration
+	ECBoostrapTimestamp time.Time
+}
+
+// Manifest identifies the specific configuration for
+// the F3 instance currently running.
+type Manifest struct {
```

Comment:
Dumb question: Why both `ManifestProvider` interface with  verbose per-field API, and `Manifest` struct? Could `ManifestProvider` be a simple one liner interface API that returns `Manifest`?

Reply:
Good catch, thanks! Initially (before the PRs for EC and other dependencies), I was thinking that for testing providers may need different underlying manifests (so the `StaticProvider` would use a different underlying Manifest that is not the same to the one used by the `DynamicProvider`). It is clear now that this won't be the case, so I will go with your suggestion of directly exposing the Manifest. Thanks!